slatedb/src/db_transaction.rs

```diff
@@ -52,7 +52,7 @@ pub struct DBTransaction {
     /// The write batch of the transaction, which contains the uncommitted writes.
     /// Users can read data from the write batch during the transaction,
     /// thus providing an MVCC view of the database.
-    write_batch: WriteBatch,
+    write_batch: RwLock<WriteBatch>,
```

Comment:
I'm not crazy about using a lock here. Granted, it will only impact transactions. But why did you opt for this vs. using `mut` as before?

Reply:
to fulfill Hussein’s requirement, interior mutability is essential. to achieve interior mutability, `RefCell` or a `Lock` are the only viable options. however, it's very easy to fail compile when using `RefCell` with code with `async` 😭.